### PR TITLE
added toleration support and reconcile fix

### PR DESCRIPTION
**Pull request by xrajesh**
Signed-off-by: Xavier Dharmaiyan <xavier@redhat.com>

**Related Issue:**  stolostron/backlog#<ISSUE_NUMBER>

### CODE REVIEW

File: api/v1alpha1/search_types.go [hunk 1]
```diff
@@ -69,6 +69,10 @@ type SearchSpec struct {
 	// +optional
 	// NodeSelector to schedule on nodes with matching labels
 	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
+
+	// +optional
+	// Tolerations to schedule pods with matching taints
```
Comment (jlpadilla) [inline api/v1alpha1/search_types.go:74]:
```suggestion
	// Tolerations to schedule pods in nodes with matching taints
```

File: controllers/common.go [hunk 2]
```diff
@@ -90,6 +90,14 @@ func getNodeSelector(deploymentName string, instance *searchv1alpha1.Search) map
 	return result
 }
 
+func getTolerations(deploymentName string, instance *searchv1alpha1.Search) []corev1.Toleration {
+	if instance.Spec.Tolerations != nil {
+		return instance.Spec.Tolerations
+	}
+	var result []corev1.Toleration
+	return result
```
Comment (jlpadilla) [inline controllers/common.go:98]: Could be a single line
```suggestion
	return []corev1.Toleration{}
```

**Review by jlpadilla — COMMENTED** [review]
/lgtm
/hold

Only minor comments or questions for personal knowledge. @xrajesh Feel free to merge after reviewing my comments.

File: api/v1alpha1/search_types.go (hunk 1 above)
Comment (xrajesh) [reply api/v1alpha1/search_types.go:74]: Done !

File: controllers/common.go (hunk 2 above)
Comment (xrajesh) [reply controllers/common.go:98]: Done

File: controllers/search_controller.go [hunk 3]
```diff
@@ -191,7 +197,14 @@ func (r *SearchReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 func (r *SearchReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&searchv1alpha1.Search{}).
-		Owns(&appsv1.Deployment{}).
+		Watches(&source.Kind{Type: &appsv1.Deployment{}}, &handler.EnqueueRequestForObject{}, builder.WithPredicates(predicate.Funcs{
+			CreateFunc: func(e event.CreateEvent) bool {
+				return false
+			},
+			UpdateFunc: func(e event.UpdateEvent) bool {
+				return false
```
[the remark this answers is not in the document]
Comment (jlpadilla) [reply controllers/search_controller.go]: ohh, now it makes sense. thank you!

**Pull request merged**